pkg/storage/tsdb/active_series.go

```diff
@@ -0,0 +1,168 @@
+package tsdb
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+const (
+	numActiveSeriesStripes = 512
+)
+
+// ActiveSeries is keeping track of recently active series for a single tenant.
+type ActiveSeries struct {
+	stripes [numActiveSeriesStripes]activeSeriesStripe
+}
+
+// activeSeriesStripe holds a subset of the series timestamps for a single tenant.
+type activeSeriesStripe struct {
+	refsMu        sync.Mutex
```

Comment:
It's called `refsMu` but doesn't project `ref` only. I would rename for clarity.